File: redsheet-portal/src/app/generator-models/generator-model.ts
import { required, maxLength, range, } from '@rx/annotations';

export class GeneratorModels {
    constructor(generatorModels?: GeneratorModels) {
        let properties = ["generatorModelId", "generatorModelName","generatorModelType","isDataAuditable","isRecordAuditable","isSoftDelete","isAdditionalValidation","isDateConversion"];
        for (let property of properties)
            if (generatorModels && generatorModels[property])
                this[property] = generatorModels[property];
    }

    generatorModelId: number = 0;

    @required()
    @maxLength(100)
    generatorModelName: string = undefined;

    @required()
    @maxLength(10)
    generatorModelType: string = undefined;

    @required()
    isDataAuditable : boolean = false;

    @required()
    isRecordAuditable : boolean = false;

    @required()
    isSoftDelete:boolean=false;

    @required()
    isAdditionalValidation:boolean=false;

    @required()
    isDateConversion:boolean = false;
    
}

File: RedSheet.DbEntities/GeneratorDbModels/GeneratorModel.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
namespace RedSheet.DbEntities.GeneratorDbModels
{
    [Table("GeneratorModels")]
    [Serializable]
    public partial class GeneratorModel 
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public int GeneratorModelId { get; set; }

        [Required]
        [MaxLength(100)]
        public string GeneratorModelName { get; set; }

        [Required]
        [MaxLength(10)]
        public string GeneratorModelType { get; set; }

        [Required]
        public bool IsDataAuditable { get; set; }

        [Required]
        public bool IsRecordAuditable { get; set; }

        [Required]
        public bool IsSoftDelete { get; set; }

        [Required]
        public bool IsAdditionalValidation { get; set; }

        [Required]
        public bool IsDateConversion { get; set; }

    }
}
